tests/handlers/auth_test.py

```diff
@@ -3,55 +3,126 @@
 from __future__ import annotations
 
 import base64
+import re
 import time
+from datetime import datetime, timedelta, timezone
 from typing import TYPE_CHECKING
 from unittest.mock import ANY
 
 import jwt
 
 from gafaelfawr.constants import ALGORITHM
+from gafaelfawr.handlers.util import AuthChallenge, AuthError, AuthType
 
 if TYPE_CHECKING:
     from tests.setup import SetupTestCallable
 
 
+def parse_www_authenticate(header: str) -> AuthChallenge:
+    """Parse a WWW-Authenticate header into this representation."""
+    auth_type_name, info = header.split(None, 1)
+    auth_type = AuthType[auth_type_name]
+
+    error = None
+    error_description = None
+    scope = None
+    for attribute in re.finditer(r'(?:\A|,\s*)([^ "=]+)="([^"]+)"', info):
```

Comment:
I get this, but I speak leet... maybe a comment here :)

Reply:
Yeah, good call. That regex turned out to be more complex than I was hoping it would be.